File: src/main/java/io/beaniejoy/dongnecafe/cafe/service/CafeService.java
package io.beaniejoy.dongnecafe.cafe.service;

import io.beaniejoy.dongnecafe.cafe.domain.Cafe;
import io.beaniejoy.dongnecafe.cafe.dto.CafeResponseDto;
import io.beaniejoy.dongnecafe.cafe.error.CafeNotFoundException;
import io.beaniejoy.dongnecafe.cafe.repository.CafeRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class CafeService {

    private final CafeRepository cafeRepository;

    public List<CafeResponseDto> getCafeList(Pageable pageable) {

        Page<Cafe> cafeListWithPagination = cafeRepository.findAll(pageable);

        return cafeListWithPagination.stream()
                .map(Cafe::toResponseDto)
                .collect(Collectors.toList());
    }

    public CafeResponseDto getCafeByCafeId(UUID cafeId) {
        Cafe cafe = cafeRepository.findById(cafeId)
                .orElseThrow(() -> new CafeNotFoundException(cafeId));

        return cafe.toResponseDto();
    }
}

File: domain/src/main/kotlin/io/beaniejoy/dongnecafe/cafe/service/CafeService.kt
package io.beaniejoy.dongnecafe.cafe.service

import io.beaniejoy.dongnecafe.cafe.model.CafeCommand
import io.beaniejoy.dongnecafe.cafe.model.CafeInfo
import io.beaniejoy.dongnecafe.cafe.model.CafeQuery
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CafeService {

    fun registerCafe(command: CafeCommand.RegisterCafe): CafeInfo.RegisteredCafe

    fun searchCafes(param: CafeQuery.SearchCafesParam, pageable: Pageable): Page<CafeInfo.CafeSearchInfo>

    fun getDetailedCafe(id: Long): CafeInfo.CafeDetailedInfo

    fun updateCafe(id: Long, command: CafeCommand.UpdateCafe)
}